java/src/test/java/com/saucelabs/simplesauce/SauceSessionTest.java

```diff
@@ -1,143 +1,222 @@
+    public void startThrowsErrorWithoutAccessKey() throws MalformedURLException {
+        environmentVariables.clear("SAUCE_ACCESS_KEY");
+        sauceSession = new SauceSession();
+
+        try {
+            sauceSession.start();
+            fail("Expected a SauceEnvironmentVariablesNotSetException to be thrown");
+        } catch (SauceEnvironmentVariablesNotSetException exception) {
+            assertEquals("Sauce Access Key was not provided", exception.getMessage());
+        }
     }
 
+    // TODO: This needs to get mocked
     @Test
-    public void sauceOptions_startWithChrome_startsChrome() {
-        dummyRemoteDriver = mock(SauceRemoteDriver.class);
-        options = new SauceOptions();
-        options.withChrome();
+    public void startsSessionAndReturnsSeleniumDriver() throws MalformedURLException {
+        sauceSession = new SauceSession();
+        sauceSession.setSauceUsername(actualUsername);
+        sauceSession.setSauceAccessKey(actualAccessKey);
 
-        sauce = new SauceSession(options, dummyRemoteDriver, dummyEnvironmentManager);
-        sauce.start();
+        WebDriver driver = sauceSession.start();
+        assertNotNull(driver);
 
-        String actualBrowser = sauce.getCurrentSessionCapabilities().getBrowserName();
-        assertEquals("chrome", actualBrowser);
+        driver.quit();
     }
 
-    @Test(expected = NullPointerException.class)
-    public void stop_newWebDriverInstanceSetByStart_stopsSession() {
-        sauce = new SauceSession(dummyRemoteDriver, dummyEnvironmentManager);
-
-        WebDriver driver = sauce.start();
-        sauce.stop();
-
+    // TODO: This needs to get mocked
+    @Test
+    public void startsSessionWithProvidedOptions() throws MalformedURLException {
+        SauceOptions sauceOptions = new SauceOptions();
+        sauceOptions.setBrowserName("firefox");
+        sauceOptions.setPlatformName("macOS 10.14");
+        sauceOptions.setBrowserVersion("68");
+
+        sauceSession = new SauceSession(sauceOptions);
+        sauceSession.setSauceUsername(actualUsername);
+        sauceSession.setSauceAccessKey(actualAccessKey);
+
+        RemoteWebDriver driver = (RemoteWebDriver) sauceSession.start();
+        MutableCapabilities capabilities = (MutableCapabilities) driver.getCapabilities();
+
+        assertEquals("firefox", capabilities.getBrowserName());
+        assertEquals("68.0", capabilities.getCapability("browserVersion"));
+        assertEquals(Platform.MAC, capabilities.getCapability("platformName"));
         driver.quit();
     }
 
+    // TODO: This needs to get mocked
     @Test
-    @Ignore("Not sure how to make this work with Mockito. To make sure that the .quit() is actually called on the webDriver")
-    public void stop_noParams_callsDriverQuit() {
-        WebDriver mockDriver = mock(WebDriver.class);
+    public void stopsSessionWithPassingResult() throws MalformedURLException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4721029)